frontend/src/components/BotList.svelte

```diff
@@ -209,8 +224,10 @@ function handleDndFinalize(e: any) {
 function handleBotClick(bot: DraggablePlayer) {
   const newId = `${bot.id}-${Math.round(Math.random() * 100000)}`;
   const idx = filteredBots.findIndex((item) => item.id === bot.id);
-  //@ts-ignore
-  filteredBots.splice(idx, 1, { ...filteredBots[idx], id: newId });
+  if (idx !== -1) {
+    // @ts-ignore
+    filteredBots[idx] = { ...filteredBots[idx], id: newId };
+  }
```

Comment:
i dont see why this would cause any bugs but its the only other thing touching the dnd logic

Reply:
reverting doesn't fix